Sources/KSCrashHandler.swift

```diff
@@ -159,9 +85,95 @@ internal class KSCrashHandler: CrashHandler {
 			userInfo[keyBreadcrumbsSerialized] = breadcrumbsSerialized
 		}
 
-		installation?.userInfo = userInfo
+		KSCrash.sharedInstance().userInfo = userInfo
+	}
+
+}
+
+class KSCrashSentryInstallation: KSCrashInstallation {
+	
+	private let generatedEvent: GeneratedEvent
+	
+	init(generatedEvent: GeneratedEvent) {
+		self.generatedEvent = generatedEvent
+		super.init(requiredProperties: [])
 	}
+	
+	override func sink() -> KSCrashReportFilter! {
+		return KSCrashReportSinkSentry(generatedEvent: generatedEvent)
+	}
+	
+}
 
+class KSCrashReportSinkSentry: NSObject, KSCrashReportFilter {
+	
+	private let generatedEvent: GeneratedEvent
+	
+	init(generatedEvent: GeneratedEvent) {
+		self.generatedEvent = generatedEvent
+		super.init()
+	}
+	
+	func filterReports(reports: [AnyObject]!, onCompletion: KSCrashReportFilterCompletion!) {
+		
+		// Mapping reports
+		let events = reports?
+			.flatMap({$0 as? CrashDictionary})
+			.map({mapReportToEvent($0)})
+		
+		// Propigating this generated event up so the SentryClient object can send it off
+		for event in events ?? [] {
+			generatedEvent(event: event)
+		}
+		
+		onCompletion?(reports, true, nil)
```

Comment:
I can see why it feels a bit weird. I didn't want to duplicate any of the sending logic and didn't want to pass a `SentryClient` object into here originally (even though it would probably look cleaner now that I think about it). The events though are send asynchronously so I would have to do a little bit of waiting on those to return in order to properly report information which would be a good idea in the end.

Thanks for listening to me thinking out loud 😊  I will make the change